python/cog/server/http.py

```diff
@@ -88,16 +89,42 @@ def create_app(
 
     predictor_ref = get_predictor_ref(config, mode)
 
+    try:
+        # TODO: avoid loading predictor code in this process
+        predictor = load_predictor_from_ref(predictor_ref)
+        InputType = get_input_type(predictor)
+        OutputType = get_output_type(predictor)
+    except Exception as ex:
+        app.state.health = Health.SETUP_FAILED
+        response = schema.PredictionResponse(input={},
+                                             output=None,
+                                             id=predictor_ref,
+                                             version=None,
+                                             created_at=datetime.now(),
+                                             started_at=datetime.now(),
+                                             completed_at=None,
+                                             logs="",
+                                             error=repr(ex),
```

Comment:
In fact, this is further evidence that returning a `PredictionResponse` from `setup` is a bad idea. `replicate-web` (the ultimate recipient of this payload) [doesn't know anything about the `error` field for the value returned as the "setup result payload", it only knows about `logs`](https://github.com/replicate/replicate-web/blob/baacbe21b83d6cfd9e6f1a850e0f5c73d6d100e8/replicate_web/models/api_serializers.py#L74).

Concretely that means that here we should be adding details on the exception to `logs`, not `error`, but I do think this is terribly confusing and we should not be returning a `PredictionResponse` from setup.

I propose we open a separate PR to revert that change.

Reply:
cc: @yorickvP